File: web/src/types/export-job.ts
/** 下载中心 / 导出任务类型。 */

export type ExportJobStatus = "queued" | "running" | "done" | "failed";

export type ExportJobRecord = {
  id: number;
  source_type: string;
  source_label: string;
  status: ExportJobStatus;
  progress_current: number;
  progress_total: number;
  filename: string;
  content_type: string;
  file_size: number;
  row_count: number;
  error_message: string;
  public_url: string;
  expires_at: string | null;
  read_at: string | null;
  created_at: string | null;
  updated_at: string | null;
};

export type PaginatedExportJob = {
  items: ExportJobRecord[];
  total: number;
  page: number;
  page_size: number;
};

export type ExportJobCreateResult = {
  job_id: number;
  message: string;
};

export type ExportJobBadge = {
  active_count: number;
  done_unread_count: number;
};

export type ExportJobMarkReadResult = {
  marked: number;
};

export const EXPORT_JOB_STATUS_LABEL: Record<ExportJobStatus, string> = {
  queued: "排队中",
  running: "导出中",
  done: "已完成",
  failed: "失败",
};

File: src/omni_api/schemas/export_job.py
"""导出任务（下载中心）DTO。"""

from __future__ import annotations

from typing import Literal

from pydantic import BaseModel

from omni_api.schemas.utc_datetime import UtcDateTime

ExportJobStatus = Literal["queued", "running", "done", "failed"]


class ExportJobCreateResult(BaseModel):
    job_id: int
    message: str = "已加入下载中心"


class ExportJobBadge(BaseModel):
    """顶栏角标：进行中（蓝）与已完成未读（绿）；展示时绿优先。"""

    active_count: int = 0
    done_unread_count: int = 0


class ExportJobMarkReadResult(BaseModel):
    marked: int = 0


class ExportJobRecord(BaseModel):
    id: int
    source_type: str
    source_label: str = ""
    status: ExportJobStatus
    progress_current: int = 0
    progress_total: int = 0
    filename: str = ""
    content_type: str = ""
    file_size: int = 0
    row_count: int = 0
    error_message: str = ""
    public_url: str = ""
    expires_at: UtcDateTime | None = None
    read_at: UtcDateTime | None = None
    created_at: UtcDateTime | None = None
    updated_at: UtcDateTime | None = None


class PaginatedExportJob(BaseModel):
    items: list[ExportJobRecord]
    total: int
    page: int
    page_size: int
